src/foam/core/SimpleFacetManager.java

```diff
@@ -15,17 +18,44 @@ public Object getInstanceOf(Object value, Class type, X x) {
   }
 
   public <T> T create(Class<T> type, X x) {
+    return create(type, Collections.<String, Object>emptyMap(), x);
+  }
+
+  public <T> T create(Class<T> type, Map<String, Object> args, X x) {
     try {
       // Automatically load FooImpl if Foo is abstract.
       // KGR: Why/where do we do this?
       if ( java.lang.reflect.Modifier.isAbstract(type.getModifiers()) ) {
         type = (Class<T>) Class.forName(type.getName() + "Impl");
       }
 
+      try {
+        java.lang.reflect.Method method = type.getMethod("getOwnClassInfo");
+        ClassInfo classInfo = (ClassInfo)method.invoke(null);
+
+        Object f = null;
+        if ( x.get(classInfo.getId() + "_Factory") != null ) {
```

Comment:
Add comments to explain what's going on.

Reply:
Done. Let me know if it's still confusing.